bst.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python
+
+
+class Node(object):
+    def __init__(self, value, left=None, right=None):
+        self.val = value
+        self.left = None
+        self.right = None
+
+    def _get_dot(self):
+        """recursively prepare a dot graph entry for this node."""
+        if self.left is not None:
+            yield "\t%s -> %s;" % (self.val, self.left.val)
+            for i in self.left._get_dot():
+                yield i
+        elif self.right is not None:
+            r = random.randint(0, 1e9)
+            yield "\tnull%s [shape=point];" % r
+            yield "\t%s -> null%s;" % (self.val, r)
+        if self.right is not None:
+            yield "\t%s -> %s;" % (self.val, self.right.val)
+            for i in self.right._get_dot():
+                yield i
+        elif self.left is not None:
+            r = random.randint(0, 1e9)
+            yield "\tnull%s [shape=point];" % r
+            yield "\t%s -> null%s;" % (self.val, r)
+
+
+class BinarySearchTree(object):
+
+    def __init__(self):
+        self.root = None
+        self.treesize = set()
+
+    def insert(self, val):
+        """insert new unique value into tree"""
+        self.treesize.add(val)
+        if self.root is None:
+            self.root = Node(val)
+        else:
+            self._insert(self.root, val)
+
+    def _insert(self, node, val):
+        if val <= node.val:
+            if node.left:
+                self._insert(node.left, val)
+            else:
+                node.left = Node(val)
+        elif val > node.val:
+            if node.right:
+                self._insert(node.right, val)
+            else:
+                node.right = Node(val)
+
+    def depth(self):
+        """return maximum depth of tree"""
+        return self._depth_helper(self.root)
+
+    def balance(self):
+        """return integer indicating level of balance based on depth of
+        each side"""
+        return self._depth_helper(self.root.right) -\
+            self._depth_helper(self.root.left)
```

Comment:
This is a place where a few temporary symbols would go a long way in helping the readability of your code.  The line continuation character is real, but you shouldn't use it unless you have to.